src/ServiceControlInstaller.Engine/FileSystem/FileUtils.cs

```diff
@@ -44,22 +79,45 @@ public static void DeleteDirectory(string path, bool recursive, bool contentsOnl
                 }
 
                 var fi = new FileInfo(f);
-                fi.Attributes &= ~FileAttributes.ReadOnly;
-                fi.Attributes &= ~FileAttributes.System;
-                fi.Delete();
+
+                RunWithRetries(() =>
+                {
+                    fi.Attributes &= ~(FileAttributes.ReadOnly | FileAttributes.System);
+                    fi.Delete();
+                });
             }
 
             if (contentsOnly)
             {
                 return;
             }
 
-            var di = new DirectoryInfo(path);
-            if (di.Exists)
+            RunWithRetries(() =>
             {
-                di.Attributes &= ~FileAttributes.ReadOnly;
-                di.Attributes &= ~FileAttributes.System;
+                di.Attributes &= ~(FileAttributes.ReadOnly | FileAttributes.System);
                 di.Delete();
+            });
+        }
+
+        public static void CloneDirectory(string srcDir, string destDir, params string[] includes)
+        {
+            Directory.CreateDirectory(destDir);
+            var files = Directory.EnumerateFiles(srcDir, "*", SearchOption.AllDirectories);
+
+            foreach (var srcFile in files)
+            {
+                var filename = Path.GetFileName(srcFile);
+                var isMatch = includes.Any(p => string.Equals(p, filename, StringComparison.OrdinalIgnoreCase));
+
+                if (isMatch)
+                {
+                    var relativePath = srcFile.Substring(srcDir.Length + 1);
+                    var destFile = Path.Combine(destDir, relativePath);
+                    var destFileDir = Path.GetDirectoryName(destFile);
+                    Trace.WriteLine($"Cloning {srcFile} to {destFile}");
```

Comment:
Do we need this at all?
Should this be `Log.Debug` ?

Reply:
Can be removed, used that during testing.
```suggestion
```